pkg/dao/types/workflow.go

```diff
@@ -83,26 +86,46 @@ func NewWorkflowStepApprovalSpec(spec map[string]any) (*WorkflowStepApprovalSpec
 		return nil, errors.New("invalid input: invalid approval type")
 	}
 
-	if v, ok := spec[WorkflowStepApprovalUsers].([]any); ok {
-		users, err := toObjectIDs(v)
-		if err != nil {
-			return nil, err
-		}
-		s.ApprovalUsers = removeDuplicatedUsers(users)
+	userIndexes := []string{
+		WorkflowStepApprovalUsers,
+		WorkflowStepApprovedUsers,
+		WorkflowStepRejectedUsers,
 	}
 
-	if v, ok := spec[WorkflowStepApprovedUsers].([]any); ok {
-		users, err := toObjectIDs(v)
-		if err != nil {
-			return nil, err
+	for i := range userIndexes {
+		if _, ok := spec[userIndexes[i]]; !ok {
+			continue
+		}
+
+		if v, ok := spec[userIndexes[i]].([]any); ok {
+			users, err := toObjectIDs(v)
```

Comment:
could you merge func toObjectIDs and removeDuplicatedUsers, since they always used together

Reply:
updated